app/src/main/java/br/org/eldorado/millelds/ui/adapter/ListAdapter.kt

```diff
@@ -0,0 +1,38 @@
+package br.org.eldorado.millelds.ui.adapter
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import br.org.eldorado.millelds.databinding.ListItemBinding
+import br.org.eldorado.millelds.model.Product
+
+class ListAdapter(
+    private val context: Context,
+    private val dataSet: List<Product>
+) : RecyclerView.Adapter<ListAdapter.ViewHolder>() {
```

Comment:
There are some points that i like to discuss here:

- I think in the challenge you'll have more adapters, then the `ListAdapter` isn't a good name, i prefer to use like:  `ProductListAdapter`
- You don't need to receive a Context in construtor, because you can get the context from `parent`
- Last but not least, the name `dataSet` isn't very clear to identify the kind of data you're manipulating in this code, i recommend change the name for something more clear/semantic

Reply:
You're right. I will provide the requested changes

Reply:
Done